### PR TITLE
COMDB2_DEBUG=1 make - to enable asserts

### PR DESCRIPTION
**Pull request by nirbhayc**

### CODE REVIEW

File: main.mk [hunk 1]
```diff
@@ -89,7 +88,15 @@ CFLAGS_DEFS+=-DBB_THREADED -D_REENTRANT -D_THREAD_SAFE
 CFLAGS_DEFS+=-DPER_THREAD_MALLOC -DMONITOR_STACK
 #CFLAGS_DEFS+=-DUSE_SYS_ALLOC
 
-CPPFLAGS+=$(CFLAGS_ARCHFLAGS) $(CFLAGS_DEFS) $(CFLAGS_INC)
+COMDB2_DEBUG=1
```
Comment (nirbhayc) [inline main.mk]: This is a temporary addition in a separate commit to enable testing of debug builds by defaults.

**Comment by nirbhayc** [comment]
All failing tests passed on my local system except for sc_timepart, which seemingly timed-out.

**Comment by riverszhang89** [comment]
The sp failure looks legitimate. Please run the test on a Redhat machine.

**Comment by nirbhayc** [comment]
> The sp failure looks legitimate. Please run the test on a Redhat machine.

Tried on a couple of RHEL versions, don't see the assertion failure. I have rebeased
the branch and triggered a rebuild.

**Comment by nirbhayc** [comment]
recovertolsn.test passed locally.

**Comment by riverszhang89** [comment]
Smoke tests are failing on your branch. Would you mind taking a look?

**Comment by nirbhayc** [comment]
Looking at its history, smoke test failure seems sporadic, not specific to this branch.

**Comment by riverszhang89** [comment]
I don't think so. Smoke test passed for almost every other branch.

**Pull request merged**